config/src/config.rs

```diff
@@ -109,28 +115,35 @@ fn check_api_secret_file(chain_type: &global::ChainTypes) -> Result<(), ConfigEr
 /// Handles setup and detection of paths for wallet
 pub fn initial_setup_wallet(
 	chain_type: &global::ChainTypes,
+	data_path: Option<PathBuf>,
 ) -> Result<GlobalWalletConfig, ConfigError> {
-	check_api_secret_file(chain_type)?;
+	check_api_secret_file(chain_type, data_path.clone())?;
 	// Use config file if current directory if it exists, .grin home otherwise
 	if let Some(p) = check_config_current_dir(WALLET_CONFIG_FILE_NAME) {
 		GlobalWalletConfig::new(p.to_str().unwrap())
 	} else {
 		// Check if grin dir exists
-		let grin_path = get_grin_path(chain_type)?;
+		let grin_path = match data_path {
+			Some(p) => p,
+			None => get_grin_path(chain_type)?,
+		};
 
 		// Get path to default config file
 		let mut config_path = grin_path.clone();
 		config_path.push(WALLET_CONFIG_FILE_NAME);
 
-		// Spit it out if it doesn't exist
+		println!("CONFIG PATH: {}", config_path.to_str().unwrap());
```

Comment:
should this be `info!` ?

Reply:
Leftover debug println, thanks for catching!